src/metabase/models/collection.clj

```diff
@@ -1405,22 +1407,24 @@
   (serdes/maybe-labeled "Collection" coll :slug))
 
 (defmethod serdes/ascendants "Collection" [_ id]
-  (let [{:keys [location]} (t2/select-one :model/Collection :id id)]
-    ;; it would work returning just one, but why not return all if it's cheap
-    (set (map vector (repeat "Collection") (location-path->ids location)))))
+  (when id
+    (let [{:keys [location]} (t2/select-one :model/Collection :id id)]
+      ;; it would work returning just one, but why not return all if it's cheap
+      (set (map vector (repeat "Collection") (location-path->ids location))))))
 
 (defmethod serdes/descendants "Collection" [_model-name id]
-  (let [location    (t2/select-one-fn :location Collection :id id)
-        child-colls (set (for [child-id (t2/select-pks-set :model/Collection {:where [:and
-                                                                                      [:like :location (str location id "/%")]
-                                                                                      [:or
-                                                                                       [:not= :type trash-collection-type]
-                                                                                       [:= :type nil]]]})]
-                           ["Collection" child-id]))
+  (let [location    (when id (t2/select-one-fn :location Collection :id id))
+        child-colls (when id ;; (sanya) NOTE: I'm not sure we should not descend here, but it seems we did not before
```

Comment:
This doesn't need to be recursive; the graph traversal in the core serdes code will find the transitive closure of nested collections under this collection.

Reply:
riiight

But I'm still unsure if it should return children for a root collection. Like, it needs for consistency, but it never did before (well, we also don't have a way to specify root collection from the CLI/APIs, so maybe it's less of a question).

Reply:
just an update: it should not return children for a root collection, because it'll just export everything all the time 🤣